### ISSUE TITLE
Remove incorrect dashicon and unnecessary label from the Shortcode block

### ISSUE THREAD

**Issue by bobbingwide** [WordPress/gutenberg#6001]
## Issue Overview
This is just one of my many requirements for the Shortcode block. 
It is the most easily implementable. 

The shortcode block is currently displayed with a label consisting of the `code-editor` dashicon SVG and a text label of "Shortcode", followed by a textarea field into which the user types the shortcode / shortcodes to be expanded. 

- The dashicon is the incorrect icon - the block itself uses `shortcode` 
- The label is unnecessary - taking up a lot of space

I suggest that whole label section is removed. 

## Steps to Reproduce (for bugs)

1. Insert a Shortcode block.
2. Notice that the dashicon in the label shows `</>` where the Block's icon is `[/]`
3. Question the need for the text label "Shortcode"

## Expected Behavior
More room to type the shortcode or shortcodes.

## Current Behavior
Incorrect icon, unnecessary label.

## Possible Solution
Remove the label.

## Screenshots / Video
![image](https://user-images.githubusercontent.com/2474435/38358364-ee5e5e02-38bc-11e8-9b41-c425f082124c.png)

## Related Issues and/or PRs
Further requirements on the shortcode block will be raised. 
I have already raised #5123 and #4456.

## Todos
- [ ] Tests
- [ ] Documentation

**Comment by danielbachhuber** [comment]
PR is #6033

**Comment by afercia** [comment]
I'd just like to remind that any issue/PR that touches accessibility-related things, should be marked with the related `accessibility` label.
For other accessibility considerations, please refer to the related PR.

**Comment by danielbachhuber** [comment]
We're not going to remove the label per https://github.com/WordPress/gutenberg/pull/6033#issuecomment-380117303

> The most simple thing in this case is to keep the label. Removing it is definitely an accessibility regression compared to the previous implementation.

The block icon is now correct on the `master` branch:

<img width="773" alt="image" src="https://user-images.githubusercontent.com/36432/39127974-6d52c6d8-46bb-11e8-9ed7-ffcba3686276.png">
